src/sql/azdata.proposed.d.ts

```diff
@@ -3992,6 +3992,11 @@ declare module 'azdata' {
 		 * Does this model view editor support save?
 		 */
 		readonly supportsSave?: boolean;
+
+		/**
+		 * A custom name for this editor (To be used in icon setting etc.)
+		 */
+		readonly customResourceName?: string;
```

Comment:
resourceName is good enough to me. also please be more descriptive here so that ppl can better understand how this works and why is this necessary

Reply:
sure will explain the icon use case more... (I was not sure how else getResourceName() is used in general, that's why tried to be generic :) )